lib/baby_squeel/nodes/proxy.rb

```diff
@@ -1,8 +1,10 @@
+require 'baby_squeel/active_record/version_helper'
+
 module BabySqueel
   module Nodes
     # This proxy class allows us to quack like any arel object. When a
     # method missing is hit, we'll instantiate a new proxy object.
-    class Proxy < ActiveSupport::ProxyObject
+    class Proxy < (BabySqueel::ActiveRecord::VersionHelper.at_least_7_2? ? BasicObject : ActiveSupport::ProxyObject)
```

Comment:
Do we need a version check here or does the `BasicObject` also work in older versions?

> ActiveSupport::ProxyObject will be removed: https://github.com/rails/rails/blob/v7.2.1.1/activesupport/lib/active_support/proxy_object.rb
> MR: https://github.com/rails/rails/commit/ea7d3c5165d5c7b6d77943482471335d828047ad

Was that part importent?
```
    undef_method :==
    undef_method :equal?
```

Reply:
good catch here - the gist of this was they deprecated `ActiveSupport:Proxy` in favor of `BaseObject` - I tested `BaseObject` against 6.x and 7.x and it does seem to work just fine.